test/support/test_dynamic_load_utils.h

```diff
@@ -60,6 +60,78 @@ test_dl_initialization(const std::vector<sycl::queue>& u)
     return 0;
 }
 
+
+template <typename Policy, typename UniverseContainer>
+int
+test_unwrap(UniverseContainer u)
+{
+    //Checking if unwrap returns the same type when passed a policy
+    using my_policy_t = Policy;
+    my_policy_t p{u};
+    bool pass = true;
+    auto policy_unwrap = oneapi::dpl::experimental::unwrap(p);
+
+    if(!std::is_same_v<decltype(p),decltype(policy_unwrap)>){
+        pass=false;
+        std::cout << "ERROR: Unwrapped policy type is not equal to the actual policy type\n";
+    }
+
+
+    //Checking if unwrap returns a different type when passed a const submission
+    const auto const_submission = oneapi::dpl::experimental::submit(
+        p, [](typename oneapi::dpl::experimental::policy_traits<Policy>::resource_type e) {
+            if constexpr (std::is_same_v<
+                              typename oneapi::dpl::experimental::policy_traits<Policy>::resource_type, int>)
+                return e;
+            else
+                return typename oneapi::dpl::experimental::policy_traits<Policy>::wait_type{};
+        });
+    const auto const_submission_unwrap = oneapi::dpl::experimental::unwrap(const_submission);
+
+    if(std::is_same_v<decltype(const_submission),decltype(const_submission_unwrap)>){
```

Comment:
I think the check can be made even stronger. We know that it should NOT be the same as the submission type, but we also know that it should be the resource_type. So, I think it would be better to also check that it returns the expected type.  (This applies for all of the checks/tests).

Reply:
Fixed